### PR TITLE
Refactor count_bugs to use libmozdata

### PR DESCRIPTION
**Pull request by YBCS**
Closes https://github.com/mozilla/bugbug/issues/3676
use libmozdata instead of calling api directly

### CODE REVIEW

File: bugbug/bugzilla.py [hunk 1]
```diff
@@ -290,13 +290,14 @@ def delete_bugs(match):
...
 
-    r = utils.get_session("bugzilla").get(
-        "https://bugzilla.mozilla.org/rest/bug", params=bug_query_params
-    )
-    r.raise_for_status()
-    count = r.json()["bug_count"]
+    counts = [0]
+
+    def handler(bug):
+        counts[0] = bug["bug_count"]
```
Comment (suhaibmujahid) [inline bugbug/bugzilla.py:296]: Using a list is a nice workaround. However, for the sake of consistency, I would suggest using a dictionary instead!

```suggestion
    data = {}

    def handler(bug):
        data["bug_count"] = bug["bug_count"]
```

**Review by suhaibmujahid — COMMENTED** [review]
Thank you, @YBCS! LGTM! I just have a minor comment.

Could you please test the funcation locally?

**Comment by YBCS** [comment]
image
Hi @suhaibmujahid  I have tested it locally 
PTAL

**Review by suhaibmujahid — APPROVED** [review]
Thank you!

**Review by suhaibmujahid — APPROVED** [review]

**Pull request merged**